ui/components/app/asset-list/asset-list.js

```diff
@@ -96,7 +99,19 @@ const AssetList = ({ onClickAsset }) => {
       />
       {process.env.TOKEN_DETECTION_V2
         ? detectedTokens.length > 0 && (
-            <DetectedTokensLink setShowDetectedTokens={setShowDetectedTokens} />
+            <DetectedTokensLink
+              onClick={() => {
+                setShowDetectedTokens(true);
+                trackEvent({
+                  event: 'Clicked "Import Tokens"',
+                  category: EVENT.CATEGORIES.WALLET,
+                  properties: {
+                    token_quantity: detectedTokens.length,
+                    tokens: detectedTokensDetails,
+                  },
+                });
+              }}
+            />
```

Comment:
Function above looks much bigger now, it will be useful to extract it out I think and not have it inline, also will be nice to have test coverage.